lib/core/presentation/pages/event/event_control_panel_page/sub_pages/event_cohosts_page/event_cohosts_page.dart

```diff
@@ -0,0 +1,167 @@
+    final t = Translations.of(context);
+    return Scaffold(
+      appBar: LemonAppBar(
+        backgroundColor: colorScheme.onPrimaryContainer,
+        title: t.event.configuration.coHosts,
+      ),
+      backgroundColor: colorScheme.onPrimaryContainer,
+      resizeToAvoidBottomInset: true,
+      body: _buildContent(),
+    );
+  }
+
+  Widget _buildContent() {
+    final t = Translations.of(context);
+    List<EventCohostRequest> eventCohostsRequests =
+        context.watch<EventDetailCohostsBloc>().state.maybeWhen(
+              fetched: (eventCohostsRequests) => eventCohostsRequests,
+              orElse: () => [],
+            );
+    return BlocListener<ManageEventCohostRequestsBloc,
+        ManageEventCohostRequestsState>(
+      listener: (context, state) {
+        state.maybeWhen(
+          orElse: () => null,
+          success: () async {
+            context.read<EventDetailCohostsBloc>().add(
+                  EventDetailCohostsEvent.fetch(
+                    eventId: widget.event?.id ?? '',
+                  ),
+                );
+          },
+        );
+      },
+      child: Padding(
+        padding: EdgeInsets.symmetric(
+          horizontal: Spacing.small,
+          vertical: Spacing.small,
+        ),
+        child: Column(
+          crossAxisAlignment: CrossAxisAlignment.start,
+          children: [
+            Text(
+              t.common.pending,
+              style: Typo.mediumPlus,
+            ),
+            SizedBox(
+              height: Spacing.smMedium,
+            ),
+            Expanded(
+              child: CohostsList(
+                eventCohostsRequests: eventCohostsRequests,
+                event: widget.event,
+              ),
+            ),
+            _buildAddCohostsButton(),
+          ],
+        ),
+      ),
+    );
+  }
+
+  _buildAddCohostsButton() {
+    return BlocBuilder<EditEventDetailBloc, EditEventDetailState>(
+      builder: (context, state) {
+        return LinearGradientButton(
+          label: t.event.cohosts.addCohosts,
+          height: 48.h,
+          radius: BorderRadius.circular(24),
+          textStyle: Typo.medium.copyWith(),
+          mode: GradientButtonMode.lavenderMode,
+          onTap: () {
+            AutoRouter.of(context).navigate(const EventAddCohostsRoute());
+          },
+          loadingWhen: state.status == EditEventDetailBlocStatus.loading,
+        );
+      },
+    );
+  }
+}
+
+class CohostsList extends StatelessWidget {
+  const CohostsList({
+    super.key,
+    required this.eventCohostsRequests,
+    required this.event,
+  });
+
+  final List<EventCohostRequest> eventCohostsRequests;
+  final Event? event;
+  @override
+  Widget build(BuildContext context) {
+    return ListView.separated(
+      padding: EdgeInsets.only(bottom: Spacing.medium),
+      shrinkWrap: true,
+      itemBuilder: (context, index) => EventCohostItem(
+        user: User.fromDto(
+          UserDto.fromJson(
```

Comment:
Why these transformation happens on UI layer ? This should only put in repository impl

Reply:
Define a type EventCohostRequest, which has `toExpanded` type is User